dojo/tools/whitehat_sentinel/parser.py

```diff
@@ -0,0 +1,253 @@
+        """
+        Manually converts the solution HTML to Markdown to avoid importing yet-another-library like Markdownify
+        Args:
+            whitehat_sentinel_vuln_solution:
+
+        Returns:
+
+        """
+        solution_html = whitehat_sentinel_vuln_solution['solution']
+
+        solution_text = re.sub(r'<.+>', '', solution_html)
+
+        solution_text = solution_text.split('References')[0]
+
+        if whitehat_sentinel_vuln_solution.get('solution_prepend'):
+            solution_text = f"{solution_text}" \
+                            f"\n {whitehat_sentinel_vuln_solution.get('solution_prepend')}"
+
+        return solution_text
+
+    def __get_href_url(self, text_to_search):
+        """
+        Searches for the anchor targets within a string that includes an anchor tag.
+        Args:
+            text_to_search: The text string to search for an anchor tag
+        Returns:
+        """
+
+        links = ''
+
+        for match in re.findall(r'(<a href=")(https://\S+)">', text_to_search):
+            links = f'{match[1]}\n{links}'
+        return links
+
+    def __remove_paragraph_tags(self, html_string):
+        """
+        Manually remove <p> tags from HTML strings to avoid importing yet-another-library.
+        Args:
+            html_string: The HMTL string to remove <p> </p> tags from
+        Returns: The original string stipped of paragraph tags
+        """
+
+        return re.sub(r'<p>|</p>', '', html_string)
+
+    def _convert_attack_vectors_to_endpoints(self, attack_vectors: List[dict]) -> List['Endpoint']:
+        """
+        Takes a list of Attack Vectors dictionaries from the WhiteHat vuln API and converts them to Defect Dojo
+        Endpoints
+        Args:
+            attack_vectors: The list of Attack Vector dictionaries
+        Returns: A list of Defect Dojo Endpoints
+        """
+
+        endpoints_list = []
+
+        # This should be in the Endpoint class should it not?
+        for attack_vector in attack_vectors:
+            try:
+                url = attack_vector['request']['url']
+                parsed_url = urlparse(url)
+                protocol = parsed_url.scheme
+                query = parsed_url.query
+                fragment = parsed_url.fragment
+                path = parsed_url.path
+                port = parsed_url.port
+                try:
+                    host, port = parsed_url.netloc.split(':')
+                except ValueError:
+                    host = parsed_url.netloc
+
+                endpoints_list.append(Endpoint(host=host,
+                                               port=port,
+                                               path=path,
+                                               protocol=protocol,
+                                               query=query,
+                                               fragment=fragment)
+                                      )
+            except Exception:
+                pass
```

Comment:
hah did you catch my comment at 159? 

        # This should be in the Endpoint class should it not?

Looks like that's fixed :)

Reply:
@damiencarol is the fix pushed to dev? I don't see that method in the Endpoint class.

Reply:
yes it is. I made a mistake, it was  `Endpoint.from_uri()` instead of `Endpoint.from_url()`

https://github.com/DefectDojo/django-DefectDojo/blob/9dba62aedc42b58c418e77c9e6d0c818c9cd1c43/dojo/models.py#L1325-L1347